modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/SidebarPanel.js

```diff
@@ -12,28 +12,96 @@
  * details.
  */
 
+import ClayAlert from '@clayui/alert';
 import ClayLoadingIndicator from '@clayui/loading-indicator';
-import React, {useImperativeHandle, useState} from 'react';
+import {fetch} from 'frontend-js-web';
+import React, {
+	useCallback,
+	useEffect,
+	useImperativeHandle,
+	useRef,
+	useState,
+} from 'react';
 
 import Sidebar from './components/Sidebar';
 
-const SidebarPanel = React.forwardRef(({onClose}, ref) => {
+const SidebarPanel = React.forwardRef(({View, fetchURL, onClose}, ref) => {
+	const CurrentView = useRef(View);
+
+	const [error, setError] = useState();
+	const [isLoading, setIsLoading] = useState();
 	const [isOpen, setIsOpen] = useState(true);
+	const [resourceData, setResourceData] = useState();
+
+	const getData = useCallback((fetchURL) => {
+		setIsLoading(true);
+		setResourceData(null);
+
+		fetch(fetchURL, {
+			method: 'GET',
+		})
+			.then((response) => response.json())
+			.then((data) => {
+				setData(data, data?.error);
+			})
+			.catch(() => {
+				setData(
+					null,
+					Liferay.Language.get('an-unexpected-error-occurred')
+				);
+			});
+	}, []);
+
+	const onCloseHandle = () => (onClose ? onClose() : setIsOpen(false));
+
+	const setData = (data, error) => {
+
+		// Force 300 ms of waiting to render the response so loading
+		// looks more natural.
+
+		setTimeout(() => {
+			setIsLoading(false);
+			setError(error);
+			setResourceData(data);
+		}, 300);
```

Comment:
nit: Maybe the use of `Promise.race()` could save the user some 300ms of waiting in some cases.